connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteThreadRunner.scala

```diff
@@ -177,6 +177,10 @@ private[connect] class ExecuteThreadRunner(executeHolder: ExecuteHolder) extends
         s"Spark Connect - ${StringUtils.abbreviate(debugString, 128)}")
       session.sparkContext.setInterruptOnCancel(true)
 
+      executeHolder.schedulerPool.foreach { pool =>
+        session.sparkContext.setLocalProperty(SparkContext.SPARK_SCHEDULER_POOL, pool)
```

Comment:
This is a core conf so can't b set runtime. This has to be set when users run the Spark Connect server.

Reply:
`conf.set("spark.scheduler.mode", "FAIR")` is a core conf, but `SparkContext.setLocalProperty(SparkContext.SPARK_SCHEDULER_POOL, pool)` is not

Reply:
It has never been tested because it was impossible for the end users to set. Does it even work?

Reply:
I commented in this api. This api should cowork with server.
![image](https://github.com/apache/spark/assets/24190835/fa43a927-f5c3-4553-9a2b-bdd2bd41bd7a)

I use this feature in our spark environment, it works.

Reply:
Server concept configuration should not be added to the client. Can you change scheduler mode at runtime?

Reply:
No, scheduler mode can't be changed at runtime. 

But I think in multi-user environment, fair scheduler and pool is better, and users need a way to set it's pool

Reply:
So, you run a Spark Connect server, and only the first Spark Session that an user creates can set the scheduler? We block all the static configurations in Spark Connect for now.